### PR TITLE
:running: Push logger further down into KubeadmControlPlane reconciliation methods

### PR DESCRIPTION
**Pull request by detiber**
**What this PR does / why we need it**:

Pushes the logger further down into the KubeadmControlPlane reconciliation methods to clean up the logging and error handling.

Related to #2242
Builds on #2379

### CODE REVIEW

File: controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go [hunk 1]
```diff
@@ -242,53 +243,28 @@ func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *
...
 		// Create new Machine w/ init
 		logger.Info("Initializing control plane", "Desired", desiredReplicas, "Existing", numMachines)
-		result, err := r.initializeControlPlane(ctx, cluster, kcp)
-		if err != nil {
-			logger.Error(err, "Failed to initialize control plane")
-			r.recorder.Eventf(kcp, corev1.EventTypeWarning, "FailedInitialization", "Failed to initialize cluster %s/%s control plane: %v", cluster.Namespace, cluster.Name, err)
-		}
-		// TODO: return the error if it is unexpected and should cause an immediate requeue
-		return result, nil
+		return r.initializeControlPlane(ctx, cluster, kcp, logger)
```
Comment (chuckha) [inline controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go]: Are we setting context that is important to this logger needing to passed down or could we instead opt for a global logger and omit the need for an additional parameter?

**Comment by chuckha** [comment]
Oh, i see what's happening. I only need to be reviewing a single commit, not both.

**Comment by detiber** [comment]
> Oh, i see what's happening. I only need to be reviewing a single commit, not both.

ah, yes, sorry. Things are a bit intertwined and this was my attempt to try and make it easier to review as well as start landing some of the changes independent of the full functionality

File: controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go (hunk 1 above)
Comment (detiber) [reply controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go]: Good point...  I think I realized this long after I started passing the logger around. I'll work on cleaning that up.

**Comment by chuckha** [comment]
no worries, a comment in the description can help, but it's not a huge deal

File: controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go (hunk 1 above)
Comment (detiber) [reply controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go]: Done.

File: controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go [hunk 2]
```diff
@@ -345,8 +323,8 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, kcp *c
 	return nil
 }
 
-func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(ctx context.Context, cluster *clusterv1.Cluster, kcp *controlplanev1.KubeadmControlPlane, requireUpgrade internal.FilterableMachineCollection) error {
-
+func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(_ context.Context, cluster *clusterv1.Cluster, kcp *controlplanev1.KubeadmControlPlane) (ctrl.Result, error) { //nolint
+	_ = r.Log.WithValues("namespace", kcp.Namespace, "kubeadmControlPlane", kcp.Name, "cluster", cluster.Name)
```
[the remark this answers is not in the document]
Comment (detiber) [reply controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go]: yeah, mostly so I don't forget when I'm in refactor hell.

**Pull request merged**